tests/nose_runner.py

```diff
@@ -93,7 +109,339 @@ def generate_tasks_input():
+    if sys.version_info < (3, 0, 0):
+        DROP_TAG = "py3"
+    else:
+        DROP_TAG = "py2"
+
+    test_file = os.path.join("tests", "cloud_answer_tests.yaml")
+    with open(test_file, 'r') as obj:
+        lines = obj.read()
+    data = '\n'.join([line for line in lines.split('\n')
+                      if DROP_TAG not in line])
+    tests = yaml.load(data)
+
+    base_argv = ['coverage', 'run', '-a', '$(which nosetests)',
+                 '--with-answer-testing', '--nologcapture',
+                 '-d', '-v', '--local', '--local-dir=%s' % ANSWER_STORE]
+    args = []
+    for answer in list(tests["answer_tests"]):
+        if tests["answer_tests"][answer] is None:
+            continue
+        argv = []
+        argv += base_argv
+        argv.append('--answer-name=%s' % "{}_{}".format(pyver, answer))
+        argv += tests["answer_tests"][answer]
+        args.append(argv)
+
+    return args
+
+
+def generate_webpage(failed_answers):
+    """Generates html for the failed answer tests
+
+    This function creates a html and embeds the images (actual, expected,
+    difference) in it for the failed answers.
+
+    Parameters
+    ----------
+    failed_answers : list of tuples (string, dict)
+        Collection of tuples where the first part is a string denoting the
+        test name, the second part is a dictionary that stores the actual,
+        expected and difference plot file locations of the test.
+
+    Returns
+    -------
+    string
+        a html page
+
+    """
+
+    html_template = """<html><head>
+    <style media="screen" type="text/css">
+    img{{
+      width:100%;
+      max-width:800px;
+    }}
+    </style>
+    <h1>{header}</h1>
+    </head><body>
+    {body}
+    </body></html>
+    """
+
+    row_template = """
+    <tr>
+    <td align="center">Actual</td>
+    <td align="center">Expected</td>
+    <td align="center">Difference</td>
+    </tr>
+    <tr>
+    <td><img src="data:image/png;base64,{0}"></td>
+    <td><img src="data:image/png;base64,{1}"></td>
+    <td><img src="data:image/png;base64,{2}"></td>
+    </tr>
+    <tr><td align="center" colspan="3"><b>{3}</b><hr></td></tr>
+    """
+
+    table_template = """<table>{rows}</table>"""
+    rows = []
+
+    for test_name, images in failed_answers:
+        encoded_images = {}
+        for key in images:
+            with open(images[key], "rb") as img:
+                img_data = base64.b64encode(img.read()).decode()
+                encoded_images[key] = img_data
+
+        formatted_row = row_template.format(encoded_images["Actual"],
+                                            encoded_images["Expected"],
+                                            encoded_images["Difference"],
+                                            test_name)
+        rows.append(formatted_row)
+
+    body = table_template.format(rows='\n'.join(rows))
+    html = html_template.format(header="Failed Answer Tests", body=body)
```

Comment:
You can combine `table_template` and `body_template` to make this one statement. (And there's no need to parameterize `header`).

Reply:
Also, maybe I'm lacking context, but I feel like 'Failed Answer Tests' isn't super descriptive. A helpful paragraph before the results might be appropriate, too.